be/src/exec/json_scanner.cpp

```diff
@@ -0,0 +1,515 @@
+            std::stringstream str_error;
+            str_error << "Invalid JsonType " << value->GetType() << ", Column Name `" << desc->col_name() << "`.";
+            return Status::RuntimeError(str_error.str());
+    }
+    return Status::OK();
+}
+
+Status JsonReader::set_tuple_value(rapidjson::Value& objectValue, Tuple* tuple, const std::vector<SlotDescriptor*>& slot_descs, MemPool* tuple_pool, bool *valid, int *nullcount) {
+    for (auto v : slot_descs) {
+        if (objectValue.HasMember(v->col_name().c_str())) {
+            rapidjson::Value& value = objectValue[v->col_name().c_str()];
+            RETURN_IF_ERROR(write_data_to_tuple(&value, v, tuple, tuple_pool));
+        } else {
+            if (v->is_nullable()) {
+                tuple->set_null(v->null_indicator_offset());
+                (*nullcount)++;
+            } else  {
+                std::stringstream str_error;
+                str_error << "The column `" << v->col_name() << "` is not nullable, but it's not found in jsondata.";
+                _state->append_error_msg_to_file("", str_error.str());
+                _counter->num_rows_filtered++;
+                *valid = false; // current row is invalid
+                break;
+            }
+        }
+    }
+    *valid = true;
+    return Status::OK();
+}
+
+/**
+ * handle input a simple json
+ * For example:
+ *  case 1. {"RECORDS": [{"colunm1":"value1", "colunm2":10}, {"colunm1":"value2", "colunm2":30}]}
+ *  case 2. {"colunm1":"value1", "colunm2":10}
+ */
+Status JsonReader::handle_simple_json(Tuple* tuple, const std::vector<SlotDescriptor*>& slot_descs, MemPool* tuple_pool, bool* eof) {
+    do {
+        bool valid = false;
+        int nullcount = 0;
+        if (_next_line >= _total_lines) {//parse json and generic document
+            RETURN_IF_ERROR(parse_json_doc(eof));
+            if (*eof) {// read all data, then return
+                return Status::OK();
+            }
+            if (_json_doc.HasMember(JsonReader::DORIS_RECORDS) && _json_doc[JsonReader::DORIS_RECORDS].IsArray() ) {
+                _total_lines = _json_doc[JsonReader::DORIS_RECORDS].Size();
+            } else {
+                _total_lines = 1; // only one row
+            }
+            _next_line = 0;
+        }
+
+        if (_json_doc.HasMember(JsonReader::DORIS_RECORDS) && _json_doc[JsonReader::DORIS_RECORDS].IsArray()) {//handle case 1
+            rapidjson::Value& values = _json_doc[JsonReader::DORIS_RECORDS];
+            rapidjson::Value& objectValue = values[_next_line];// json object
+            RETURN_IF_ERROR(set_tuple_value(objectValue, tuple, slot_descs, tuple_pool, &valid, &nullcount));
+        } else {// handle case 2
+            RETURN_IF_ERROR(set_tuple_value(_json_doc, tuple, slot_descs, tuple_pool, &valid, &nullcount));
+        }
+        _next_line++;
+        if (!valid) {
+            continue;
+        }
+        if (nullcount == slot_descs.size()) {// All fields is null, read next one
```

Comment:
Whenever you skip the line, you should print the line to the runtime state, so that user can find which line is wrong.
So I think the case `nullcount == slot_descs.size()` can be just put into the `set_tuple_value()` method.